runtime/compilers/rillv1/parser.go

```diff
@@ -246,6 +246,16 @@ func (p *Parser) IsIgnored(path string) bool {
 	return !pathIsYAML(path) && !pathIsSQL(path) && !pathIsDotEnv(path)
 }
 
+func (p *Parser) FindFilesInDir(dir string) []string {
+	var files []string
+	for path := range p.resourcesForPath {
+		if strings.HasPrefix(path, dir) {
+			files = append(files, path)
+		}
+	}
+	return files
+}
```

Comment:
1. Needs docstring
2. Needs to ensure a trailing slash on the dir

Suggestion:

```suggestion
// TrackedPathsInDir returns the paths under the given directory that the parser currently has cached results for.
func (p *Parser) TrackedPathsInDir(dir string) []string {
	// Ensure dir has a trailing path separator
	if len(dir) > 0 && dir[len(dir)-1] != '/' {
		dir += '/'
	}

	// Find paths
	var paths []string
	for path := range p.resourcesForPath {
		if strings.HasPrefix(path, dir) {
			paths = append(paths, path)
		}
	}
	return paths
}
```